airflow/contrib/hooks/spark_sql_hook.py

```diff
@@ -122,9 +122,30 @@ def run_query(self, cmd="", **kwargs):
 
         :param cmd: command to remotely execute
         :param kwargs: extra arguments to Popen (see subprocess.Popen)
-        :return: handle to subprocess
         """
         prefixed_cmd = self._prepare_command(cmd)
-        return subprocess.Popen(prefixed_cmd, 
+        self.sp = subprocess.Popen(prefixed_cmd, 
                                 stdout=subprocess.PIPE, 
                                 stderr=subprocess.PIPE)
+        # using two iterators here to support 'real-time' logging 
+        for line in iter(self.sp.stdout.readline, b''):
+            line = line.decode('utf-8').strip()
+            logging.info(line)
+        for line in iter(self.sp.stderr.readline, b''):
+            line = line.decode('utf-8').strip() 
+            logging.info(line)
+        output, stderr = self.sp.communicate()
+
+        if self.sp.returncode:
+            raise AirflowException("Cannot execute {} on {}. Error code is: "
+                                   "{}. Output: {}, Stderr: {}"
+                                   .format(cmd, self.conn.host,
+                                           self.sp.returncode, output, stderr))
+            
+    def kill(self):
+        if hasattr(self, 'sp'):
```

Comment:
I would put this in __init__. It is considered bad practice to have object variables initialized outside init.